src/components/layout/Layout.ts

```diff
@@ -136,6 +136,41 @@ export class Layout {
     }
     this.elements = [];
   }
+
+  // Duplicate element in the selected cell
+  // It will either be placed in the hoveredCell, if it is not a wall and not already occupied
+  // Or it will be placed in the first empty cell
+  duplicateElement(
+    selectedCell: [number, number],
+    hoveredCell: [number, number] | undefined,
+  ) {
+    const selectedCellY = selectedCell[0];
+    const selectedCellX = selectedCell[1];
+
+    if (!(this.layout[selectedCellY][selectedCellX] instanceof SquareType)) {
+      throw new Error('Cannot rotate a wall element');
```

Comment:
Not the right error message here, I assume you meant 'Cannot duplicate a wall element'

Reply:
Done